plugins/check_smtp.c

```diff
@@ -650,6 +651,7 @@ process_arguments (int argc, char **argv)
 #else
 			usage (_("SSL support not available - install OpenSSL and recompile"));
 #endif
+			break;
```

Comment:
Since there now two possibilities to use TLS, the `break` is correct to prevent the usage of direct TLS in every case.
However I see a slight problem here: This changes the API. If a user did not specify STARTTLS previously, but choose the certificate check option, it would initiate STARTTLS regardless.
Now the `-D` flag is silently ignored.
While this behaviour arguably "better" it will silently change existing setups.

This change actually happened in the previous PR and I did not see it. Mea culpa.

No I see two options here:
1) Restore the previous behaviour, aumatically enable STARTTLS when `-D` is used, which I do not like very much, but would preserve backward compatibility.
2) Let the plugin fail with `UNKNOWN` and inform the user, that a TLS option has to be chosen.

@sni @waja opinions?

Reply:
Sorry for having introduced this issue. Although these were just backports from nagions-plugins... :innocent: 

If I would have been asked I would prefer option 2. In general it's better to be explicit instead of implicit. This change of the interface should be announced in the `NEWS` then.

Reply:
@waja @RincewindsHat I added a check for the SSL/TLS method when a certificate check is requested. Meaning if `-D`/`--certificate` is given either `-s`/`--ssl`/`--tls` or `-S`/`--starttls` have to be set.

Please have a look at it if it suits you.

Reply:
Since @waja seems to not have the time, I say we go the safe way and keep backwards compatibility. `-D` should assume starttls if `--tls` is not set explicitely.

I would like to change this in a major release though.

Reply:
It doesn't matter, from my (package maintainers) perspective, if you break backward compatibility with a minor or a major release.

Reply:
Since this part has now been handled in a backwards-compatible way with 2eea6bb2a04bbfb, the remaining parts of this PR to allow checking validity of certs with both STARTTLS and TLS-on-connect seem straightforward